x-pack/plugins/observability_solution/observability_onboarding/public/assets/auto_detect.sh

```diff
@@ -1,5 +1,34 @@
 #!/bin/bash
 
+fail() {
+  printf "%s\n" "$@" >&2
+  exit 1
+}
+
+if [ -z "${BASH_VERSION:-}" ]; then
+  fail "Bash is requred to run this script"
+fi
+
+if [ -z "$INSTALL_API_KEY_ENCODED" ]; then
```

Comment:
What's the reason for using environmental variables for these parameters? 

I think command line arguments would be more appropriate. Environmental variables are usually only used for global parameters that are used by multiple applications and don't change between invocations (e.g. location of JDK). 

I would also try and shorten the names a bit rather than be too verbose and technical. As an example we can have a `--help` command that explains that the api key should be base64 encoded instead of having that as the parameter name. 

The shorter and less threatening the snippet the user has to copy/paste the better. 

```bash
bash auto-detect.sh --url=http://localhost:5601 --id=3ad72dec-d263-44ec-a9b3-d654fba39a18 --ingest-key=UVRJU3hvOEJGM2hXam9ISGtOY1Q6SnFNdzhUbFlRcy1QUVZoOVQ5SWxwZw== --install-key=UVRJU3hvOEJGM2hXam9ISGtOY1Q6SnFNdzhUbFlRcy1QUVZoOVQ5SWxwZw==
```

Reply:
You're right, named parameters would be the way to go, I've used env variables as a quick substitute because the current script relies on parameters order and I did not want to have that. I'll look into making them proper named arguments 👍